src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -86,12 +87,32 @@ private long getAgeInMilliSeconds(Date birthday) {
     }
 
     private int getAge(Date birthday) {
-        Calendar c = Calendar.getInstance();
-        c.setTime(new Date());
-        int currentYear = c.get(Calendar.YEAR);
-        c.setTime(birthday);
-        int birthYear = c.get(Calendar.YEAR);
-        return currentYear - birthYear;
+        int years;
+        int months;
+        Calendar birthDayMili = Calendar.getInstance();
+        birthDayMili.setTimeInMillis(birthday.getTime());
+        long currentTime = System.currentTimeMillis();
+        Calendar now = Calendar.getInstance();
+        now.setTimeInMillis(currentTime);
+        years = now.get(Calendar.YEAR) - birthDayMili.get(Calendar.YEAR);
+        int currMonth = now.get(Calendar.MONTH) + 1;
+        int birthMonth = birthDayMili.get(Calendar.MONTH) + 1;
+        months = currMonth - birthMonth;
+        if (months < 0)
+        {
+            years--;
+            months = 12 - birthMonth + currMonth;
+            if (now.get(Calendar.DATE) < birthDayMili.get(Calendar.DATE))
+                months--;
+        } else if (months == 0 && now.get(Calendar.DATE) < birthDayMili.get(Calendar.DATE))
+        {
+            years--;
+            months = 11;
+        }
+        if (months == 12) {
+            years++;
+        }
+        return years;
```

Comment:
Wow, I see what you tried to achieve here: because before it might have returned one year too many if your birthday is later in the year. However, I think you overthought on that. Sometimes, it's better to see what other methods we already have that you CAN use. Instead of writing 15 lines of code that might be incorrect (I didn't check). i would have done a oneliner:
`return (int) (getAgeInMilliSeconds(birthday) / Time.ONE_YEAR)`

Reply:
You can test that with 
```
Date date = new Date(0);
assert (getAge(date) == 52))
date.setMonth(8);
assert (getAge(date) == 51))
```

(But don't submit that code, because it will be broken in 3 months) ;)

Reply:
You could test it via converting date to localdate:
![image](https://user-images.githubusercontent.com/75019468/168027530-fbf3f1dc-9099-4991-a7e3-75a1045bf4fc.png)

Reply:
problem with : return (int) (getAgeInMilliSeconds(birthday) / Time.ONE_YEAR)    is that it doesn't account for 29th of februaries and so it gets slightly distorted

Reply:
Ah yes, that's true, haven't thought about that. But I still think you might want to not calculate it manually

Reply:
Yes, but i thought we dont use localdate due to the problem with the day not being correct?

Reply:
Erm, if I remember correctly that issue arose with Date, and not LocalDate (we probably should have better used LocalDate instead of Date).

Reply:
-Irrelevant with new implementation